cloudbees-cd/kubernetes/gke/prod.md

```diff
@@ -0,0 +1,256 @@
+    ```
+- Create custom storage class for CSI Filestore driver for provisioning filestore instance in the same network as the GKE cluster
+    ```bash
+    cat <<EOF | kubectl apply -f -
+    apiVersion: storage.k8s.io/v1
+    kind: StorageClass
+    metadata:
+      name: filestore-sc
+    provisioner: filestore.csi.storage.gke.io
+    parameters:
+      tier: standard
+      network: projects/$GCP_PROJECT/global/networks/$GCP_VPC_NETWORK
+    volumeBindingMode: Immediate
+    allowVolumeExpansion: true
+    EOF
+  ```
+### GCP SQL configuration
+- Create GCP SQL instance with private connection
+    ```bash
+    GCP_DB_INSTANCE_NAME=<gcp-db-instance-name>
+    DB_ROOT_PASSWORD=$(openssl rand -base64 32)
+    gcloud beta sql instances create $GCP_DB_INSTANCE_NAME \
+      --database-version=POSTGRES_13 \
+      --network=$GCP_VPC_NETWORK \
+      --no-assign-ip \
+      --no-require-ssl \
+      --storage-type=SSD \
+      --storage-size=10 \
+      --storage-auto-increase \
+      --tier=db-custom-2-7680 \
+      --zone=$GCP_ZONE \
+      --root-password=$DB_ROOT_PASSWORD \
+      --database-flags=max_connections=1000 \
+      --allocated-ip-range-name=google-managed-services-$GCP_VPC_NETWORK
+  ```
+- Create new database and user
+    ```bash
+    GCP_DB_NAME=<gcp-db-name>
+    GCP_DB_USER=<gcp-db-user>
+    GCP_DB_PASSWORD=<gcp-db-password>
+    gcloud sql databases create $GCP_DB_NAME \
+      --instance=$GCP_DB_INSTANCE_NAME
+    gcloud sql users create $GCP_DB_USER \
+      --instance=$GCP_DB_INSTANCE_NAME \
+      --password=$GCP_DB_PASSWORD
+  ```
+- Get the IP address of the database instance
+    ```bash
+    GCP_DB_IP=$(gcloud sql instances describe $GCP_DB_INSTANCE_NAME \
+      --format='value(ipAddresses.ipAddress)')
+    echo $GCP_DB_IP
+  ```
+- Create DNS zone
+    ```bash
+    GCP_DNS_ZONE=<gcp-dns-zone>                       # e.g GCP_DNS_ZONE=cd-internal
+    GCP_DNS_NAME=$GCP_DNS_ZONE.
+    gcloud dns managed-zones create $GCP_DNS_ZONE \
+      --dns-name=$GCP_DNS_NAME \
+      --description="CloudBees CD DNS zone" \
+      --visibility=private \
+      --networks=$GCP_VPC_NETWORK
+  ```
+- Create DNS record for the SQL instance
+    ```bash
+    DB_DNS_ADDRESS=$GCP_DB_INSTANCE_NAME.$GCP_DNS_NAME
+    gcloud dns record-sets create $DB_DNS_ADDRESS \
+      --zone=$GCP_DNS_ZONE \
+      --type="A" \
+      --ttl="300" \
+      --rrdatas=$GCP_DB_IP
+  ```  
+
+### CloudBees CD installation
+
+- Download production values file
+    ```bash
+    curl -fsSL -o cloudbees-cd-production.yaml https://raw.githubusercontent.com/cloudbees/cloudbees-examples/master/cloudbees-cd/kubernetes/cloudbees-cd-prod.yaml
+  ```
+- Create k8s namespace
+    ```bash
+    kubectl create namespace $NAMESPACE
+  ```
+- Create db credentials secret
+    ```bash
+    kubectl create secret generic flow-db-secret \
+      --namespace $NAMESPACE \
+      --from-literal=DB_USER=$GCP_DB_USER \
+      --from-literal=DB_PASSWORD=$GCP_DB_PASSWORD
+  ```
```

Comment:
Without server secret, I was not able to run cluster: 
Perhaps it would be ok to add the next command to create a secret:


- Create server credential secret
      ```bash
       kubectl create secret generic your-flow-admin-secret \
       --from-literal=CBF_SERVER_ADMIN_PASSWORD=1234root
      ```